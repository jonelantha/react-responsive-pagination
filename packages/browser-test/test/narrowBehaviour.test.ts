import { setupThrowOnError, stringifyWithUndefined } from './helper';

beforeAll(() => {
  setupThrowOnError(page);
});

const testWidths = [150, 200, 250];

const narrowBehaviours = [
  undefined,
  'dropNav',
  'dropEllipsis',
  'dropNavThenEllipsis',
  'dropEllipsisThenNav',
];

describe.each(narrowBehaviours)(
  'Auto sizing with narrowBehaviour %p',
  narrowBehaviour => {
    beforeAll(async () => {
      await page.goto('about:blank');

      await page.goto(`${harnessUrl}bootstrap4`);

      await page.fill(
        '#narrowBehaviourNameAsJson',
        stringifyWithUndefined(narrowBehaviour),
      );

      await page.fill('#totalAsJson', '100');
    });

    test.each(testWidths.map(width => [width]))(
      'renders correctly with viewport width %ipx',
      async width => {
        await page.setViewportSize({ width, height: 700 });

        await page.evaluate(() => new Promise(requestAnimationFrame));

        const paginationHtml = await page.$eval('ul.pagination', ul => ul.outerHTML);

        expect(paginationHtml).toMatchSnapshot();
      },
    );
  },
);
