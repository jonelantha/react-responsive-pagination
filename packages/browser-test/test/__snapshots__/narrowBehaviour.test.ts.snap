// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auto sizing with narrowBehaviour "dropEllipsis" renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour "dropEllipsis" renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour "dropEllipsis" renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour "dropNav" renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour "dropNav" renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour "dropNav" renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour "dropEllipsisThenNav" renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour "dropEllipsisThenNav" renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour "dropEllipsisThenNav" renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour "dropNavThenEllipsis" renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour "dropNavThenEllipsis" renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour "dropNavThenEllipsis" renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour ["dropEllipsis", "dropNav"] renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour ["dropEllipsis", "dropNav"] renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour ["dropEllipsis", "dropNav"] renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour ["dropNav", "dropEllipsis"] renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour ["dropNav", "dropEllipsis"] renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour ["dropNav", "dropEllipsis"] renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour undefined renders correctly with viewport width 150px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour undefined renders correctly with viewport width 200px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;

exports[`Auto sizing with narrowBehaviour undefined renders correctly with viewport width 250px 1`] = `"<ul class="pagination justify-content-center"><li class="page-item disabled"><span class="page-link" aria-label="Previous"><span aria-hidden="true">«</span></span></li><li class="page-item active" aria-current="page"><a class="page-link" href="#">1</a></li><li class="page-item"><a class="page-link" href="#">2</a></li><li class="page-item disabled" aria-hidden="true"><span class="page-link">…</span></li><li class="page-item"><a class="page-link" href="#">100</a></li><li class="page-item"><a class="page-link" href="#" aria-label="Next"><span aria-hidden="true">»</span></a></li></ul>"`;
